make save_request and save_response being called at different moments

mmdc88 opened the pull request:
## Description

Doing this rearrangement tshield would always generate the original_request from my API and I would be able to test what data my API is generating without needing to create any mock.

## Type of change
Minor - Code rearrangement

## How Has This Been Tested
As this is just a rearrangement of the code I used the existing tests to validate if everything is working properly

flaviomeira commented:
Hello @mmdc88. Thanks for your PR.

Could you add a component test in order to better describe the new behavior?

mmdc88 commented:
Hey @flaviomeira, I added the scenario to make it more explicit what my change is about.
I modified the run file to execute my scenario with the GRPC service down.
